controller/dataplane/controller_reconciler_utils.go

```diff
@@ -77,6 +84,170 @@ func (r *Reconciler) ensureDataPlaneAddressesStatus(
 	return false, nil
 }
 
+// ensureMappedConfigMapToKongPluginInstallationForDataPlane ensures that the KongPluginInstallation
+// resources referenced by the DataPlane are resolved and DataPlane is configured to use them.
+// During resolving for each DataPlane based on each instance of KongPluginInstallation
+// ConfigMap is created and mounted. The DataPlane manages its lifecycle. It returns a slice
+// of custom plugins that are intended to be used to generate a Deployment.
+func ensureMappedConfigMapToKongPluginInstallationForDataPlane(
+	ctx context.Context, logger logr.Logger, c client.Client, dataplane *operatorv1beta1.DataPlane,
+) ([]customPlugin, ctrl.Result, error) {
+	configMapsOwned, err := findCustomPluginConfigMapsOwnedByDataPlane(ctx, c, dataplane)
+	if err != nil {
+		return nil, ctrl.Result{}, err
+	}
+	configMapsToRetain := make(map[types.NamespacedName]struct{}, len(configMapsOwned))
+
+	var cps []customPlugin
+	for _, kpiNN := range dataplane.Spec.PluginsToInstall {
+		kpiNN := types.NamespacedName(kpiNN)
+		if kpiNN.Namespace == "" {
+			kpiNN.Namespace = dataplane.Namespace
+		}
+
+		cp, reconciliationResult, err := populateDedicatedConfigMapForKongPluginInstallation(
+			ctx, logger, c, configMapsOwned, kpiNN, dataplane,
+		)
+		if err != nil || reconciliationResult.Requeue {
```

Comment:
Ditto about `.RequeueAfter`.

We might also consider adding a `func` which would simply return `true` if either of those are set.